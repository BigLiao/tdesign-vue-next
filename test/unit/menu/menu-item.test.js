import { mount } from '@vue/test-utils';
import { MenuItem } from '@/src/menu/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('MenuItem', () => {
  // test props api
  describe('props', () => {
    it(':name', () => {
      const wrapper = mount({
        render() {
          return <MenuItem name="1-1"></MenuItem>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':route', () => {
      const wrapper = mount({
        render() {
          return <MenuItem route={'/user/list'}></MenuItem>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':disabled', () => {
      const wrapper = mount({
        render() {
          return <MenuItem disabled={true}></MenuItem>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('slot', () => {
    it('<icon>', () => {
      const wrapper = mount(MenuItem, {
        scopedSlots: {
          icon: '<div></div>',
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it('<default>', () => {
      const wrapper = mount(MenuItem, {
        scopedSlots: {
          default: '<div></div>',
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
