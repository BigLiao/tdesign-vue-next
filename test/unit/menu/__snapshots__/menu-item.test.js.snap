// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem props :disabled 1`] = `<li class="t-menu__item t-is-disabled"><span></span></li>`;

exports[`MenuItem props :name 1`] = `<li class="t-menu__item"><span></span></li>`;

exports[`MenuItem props :route 1`] = `<a href="" target="_self" class="t-menu__item"><span></span></a>`;

exports[`MenuItem slot <default> 1`] = `<li class="t-menu__item"><span><div></div></span></li>`;

exports[`MenuItem slot <icon> 1`] = `
<li class="t-menu__item">
  <div></div><span></span>
</li>
`;
